src/hotspot/share/prims/vectorSupport.cpp

```diff
@@ -124,7 +117,11 @@ Handle VectorSupport::allocate_vector_payload_helper(InstanceKlass* ik, frame* f
 
   int num_elem = klass2length(ik);
   BasicType elem_bt = klass2bt(ik);
-  int elem_size = type2aelembytes(elem_bt);
+  // Inserting a VectorStoreMask before stitching the mask
+  // to SafePointNode will ensure packing the mask into a
+  // byte array for masks present in both predicated register
+  // or vector registers.
+  int elem_size = is_mask ? 1 : type2aelembytes(elem_bt);
```

Comment:
If `elem_bt == T_BOOLEAN` for masks, you can get rid of `is_mask` usages in `VectorSupport::allocate_vector_payload_helper()` and just introduce a special case in `VectorSupport::klass2bt` (as already done for `VectorShuffle`s).